src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/ContentService.cs

```diff
@@ -228,14 +227,127 @@ public ContentService(ContentDbContext context)
                     case ContentBlockType.InsetTextBlock:
                         return await UpdateInsetTextBlock((InsetTextBlock) blockToUpdate, request.Heading, request.Body);
                     case ContentBlockType.DataBlock:
-                        return new Either<ValidationResult, IContentBlock>(
-                            ValidationUtils.ValidationResult(ValidationErrorMessages.IncorrectContentBlockTypeForUpdate));
+                        return ValidationResult<IContentBlock>(
+                            ValidationErrorMessages.IncorrectContentBlockTypeForUpdate);
                     default:
                         throw new ArgumentOutOfRangeException();
                 }
             });
         }
 
+        public async Task<Either<ValidationResult, List<IContentBlock>>> GetUnattachedContentBlocksAsync(Guid releaseId, ContentBlockType type)
+        {
+            var unattachedContentBlocks = await _context
+                .ReleaseContentBlocks
+                .Include(join => join.ContentBlock)
+                .Where(join => join.ReleaseId == releaseId)
+                .Select(join => join.ContentBlock)
+                .Where(contentBlock => contentBlock.ContentSectionId == null 
+                                       && contentBlock.Type == type.ToString())
+                .ToListAsync();
```

Comment:
It might be quite nice to introduce a type param here instead of using the enum.

unattachedContentBlocks.
...
OfType<T>().ToListAsync()
will give you a typed list.

The return type can be a List of the same type. 
The GetUnattachedDataBlocks method further up in this class can return a List of type DataBlocks.